contexts/currency-context.tsx

```diff
@@ -34,11 +34,14 @@ export const CurrencyProvider = ({ children }: React.PropsWithChildren<{}>) => {
 
   const { address: userAddress } = useAccount()
   const { data: balance } = useBalance({ address: userAddress })
+  const { data: currencies } = useQuery(['currencies'], async () => {
+    return fetchCurrencies()
+  })
 
   const { data: prices } = useQuery(
     ['prices'],
     async () => {
-      const currencyAddresses = (await fetchCurrencies()).map(
```

Comment:
Let's use `fetchCurrencies` here. It's better to have an extra api call if it means we can remove the dependency to the useQuery above. Code that causes side effects are bad, and code that depends on side effects are even worse. It's the path to spaghetti code.

Reply:
Reverted: https://github.com/clober-dex/coupon.finance/pull/25/commits/c198f3e9f1019719480cc8cb842d483dd62dd34b